lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java

```diff
@@ -0,0 +1,872 @@
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_ThrowException_When_CreatingTheSameTaskTwice() throws Exception {
+    Task task = createDefaultTask();
+    task.setExternalId("MyExternalId");
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask).isNotNull();
+
+    Task task2 = createDefaultTask();
+    task2.setExternalId("MyExternalId");
+
+    ThrowingCallable call = () -> taskService.createTask(task2);
+    TaskAlreadyExistException e = catchThrowableOfType(call, TaskAlreadyExistException.class);
+    assertThat(e.getExternalId()).isEqualTo("MyExternalId");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_CreateTask_When_CustomAttributesAreSpecified() throws Exception {
+    Task task = createDefaultTask();
+    Map<String, String> customAttributesForCreate = createSimpleCustomPropertyMap(13);
+    task.setCustomAttributeMap(customAttributesForCreate);
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask).isNotNull();
+    // verify that the map is correctly retrieved from the database
+    Task retrievedTask = taskService.getTask(createdTask.getId());
+    Map<String, String> customAttributesFromDb = retrievedTask.getCustomAttributeMap();
+    assertThat(customAttributesFromDb).isNotNull();
+    assertThat(customAttributesFromDb).isEqualTo(customAttributesForCreate);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_CreateTask_When_AttachmentIsSpecified() throws Exception {
+    Map<String, String> customAttributesForCreate = createSimpleCustomPropertyMap(27);
+    ClassificationSummary classificationSummary =
+        defaultTestClassification()
+            .type("DOCUMENT")
+            .category("EXTERNAL")
+            .buildAndStoreAsSummary(classificationService, "admin");
+    Task task = createDefaultTask();
+    Attachment attachment =
+        TaskAttachmentBuilder.newAttachment()
+            .objectReference(defaultObjectReference)
+            .classificationSummary(classificationSummary)
+            .created(Instant.now())
+            .modified(Instant.now())
+            .customAttributes(customAttributesForCreate)
+            .received(Instant.parse("2018-01-15T00:00:00Z"))
+            .channel("E-MAIL")
+            .build();
+    task.addAttachment(attachment);
+
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask.getId()).isNotNull();
+    assertThat(createdTask.getCreator()).isEqualTo("user-1-2");
+    assertThat(createdTask.getAttachments().get(0).getCustomAttributeMap().toString())
+        .contains(
+            IntStream.rangeClosed(1, 27)
+                .mapToObj(String::valueOf)
+                .map(
+                    number ->
+                        String.format("Property_%s=Property Value of Property_%s", number, number))
+                .collect(Collectors.toSet()));
+    Task readTask = taskService.getTask(createdTask.getId());
```

Comment:
You don't need to read the Task again. Just use createdTask in your assertions. Please also add a variable for readTask.getAttachments().get(0) for better readability